emmopy/emmocheck.py

```diff
@@ -463,7 +463,9 @@ def checker(onto, ignore_namespace):
         checker(self.onto, self.ignore_namespace)
 
 
-def main():  # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+def main(
+    argv: list = None,
```

Comment:
What is the purpose of this argument?

Reply:
Ahh, I see it now why you introduced `argv`. It is used by the tests. 

It this the preferred way to test python tools - by calling their main() function from within Python?
Wouldn't it be more direct to use subprocess to check them by calling them as they would have been called from the shell?

Reply:
If we keep argv, I think it would be good with a comment to explain why it is there.

Reply:
Sure. And I'd say that one can test it in both ways. I would test it through Python to be able to keep everything in a "closed pytest environment", but at the same time, a `multiprocessing.Process` or `submodule.run` approach would be good to have as well to make sure the shell calls work as intended.
In other words - I wouldn't remove this work, but simply add to it.

And sure - I can create a comment in the `main()` functions' doc-strings.